tests/tests_data/streaming/test_dataset.py

```diff
@@ -15,46 +15,50 @@
 from unittest import mock
 
 import pytest
-import torch
 from lightning import seed_everything
 from lightning.data.datasets.env import _DistributedEnv
 from lightning.data.streaming import Cache
-from lightning.data.streaming.dataloader import StreamingDataLoader
 from lightning.data.streaming.dataset import StreamingDataset, _try_create_cache_dir
-from lightning.data.streaming.item_loader import TokensLoader
 from lightning.data.streaming.shuffle import FullShuffle, NoShuffle
-from lightning.pytorch.demos.boring_classes import RandomDataset
 from torch.utils.data import DataLoader
 
 
 def test_streaming_dataset(tmpdir, monkeypatch):
     seed_everything(42)
 
+    dataset = StreamingDataset(input_dir=tmpdir)
     with pytest.raises(ValueError, match="The provided dataset"):
-        dataset = StreamingDataset(input_dir=tmpdir)
+        iter(dataset)
+    dataset = StreamingDataset(input_dir=tmpdir)
+    with pytest.raises(ValueError, match="The provided dataset"):
+        _ = dataset[0]
 
-    dataset = RandomDataset(128, 64)
-    dataloader = StreamingDataLoader(dataset, cache_dir=tmpdir, chunk_bytes=2 << 12)
-    for batch in dataloader:
-        assert isinstance(batch, torch.Tensor)
+    cache = Cache(tmpdir, chunk_size=10)
+    for i in range(12):
+        cache[i] = i
+    cache.done()
+    cache.merge()
 
-    dataset = StreamingDataset(input_dir=tmpdir, item_loader=TokensLoader(block_size=10))
+    dataset = StreamingDataset(input_dir=tmpdir)
 
-    assert len(dataset) == 816
```

Comment:
These values didn't make sense to me. The StreamingDataLoader() had a side effect on the iter() call. I couldn't explain it. Now I just rewrote the test without StreamingDataLoader